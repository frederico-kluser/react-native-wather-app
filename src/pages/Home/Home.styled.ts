import styled from 'styled-components/native';
import {deviceWidth} from '../../helpers/dimentions';
import {GLOBAL_PADDING} from '../../styles/theme';

export const Container = styled.View`
  padding: ${GLOBAL_PADDING}px;
  flex-direction: row;
  flex-wrap: wrap;
`;

export const Logo = styled.Image`
  left: ${deviceWidth / 2 - 100 - GLOBAL_PADDING}px;
  position: relative;
  margin-top: 350px;
  height: 200;
  width: 200;
`;
