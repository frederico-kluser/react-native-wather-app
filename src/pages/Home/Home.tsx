import React, {useEffect, useState} from 'react';
import Item from '../../components/molecule/Item/Item';
import ItemModal from '../../components/organism/ItemModal/ItemModal';
import {Container} from './Home.styled';

const Home = ({scrollY, setScroll}) => {
  const [selectedItem, setSelectedItem] = useState(-1);
  const [cities, setCities] = useState([
    {
      name: 'New York',
      temperature: '8',
      maxTemperature: '10',
      minTemperature: '2',
      status: 'cloudy',
    },
    {
      name: 'SÃ£o Paulo',
      temperature: '10',
      maxTemperature: '12',
      minTemperature: '2',
      status: 'rainy',
    },
    {
      name: 'Rio de Janeiro',
      temperature: '12',
      maxTemperature: '14',
      minTemperature: '2',
      status: 'sunny',
    },
    {
      name: 'New York',
      temperature: '8',
      maxTemperature: '10',
      minTemperature: '2',
      status: 'cloudy',
    },
    {
      name: 'Orlando',
      temperature: '10',
      maxTemperature: '12',
      minTemperature: '2',
      status: 'rainy',
    },
    {
      name: 'London',
      temperature: '12',
      maxTemperature: '14',
      minTemperature: '2',
      status: 'sunny',
    },
    {
      name: 'Tokyo',
      temperature: '8',
      maxTemperature: '10',
      minTemperature: '2',
      status: 'cloudy',
    },
    {
      name: 'Dubai',
      temperature: '10',
      maxTemperature: '12',
      minTemperature: '2',
      status: 'rainy',
    },
  ]);

  useEffect(() => {
    setScroll(selectedItem === -1);
  }, [selectedItem]);

  return (
    <Container>
      {selectedItem >= 0 && (
        <ItemModal
          index={selectedItem}
          name={cities[selectedItem].name}
          setSelectedItem={setSelectedItem}
          scrollY={scrollY}
        />
      )}
      {cities.map(({name}: any, index: number) => (
        <Item
          func={setSelectedItem}
          index={index}
          key={name + index}
          lastElement={index === cities.length - 1}
          name={name}
        />
      ))}
    </Container>
  );
};

export default Home;
