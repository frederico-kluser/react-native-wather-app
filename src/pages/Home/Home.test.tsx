import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

jest.useFakeTimers();

test('renders correctly', () => {
  const tree = renderer
    .create(<Home scrollY={0} setScroll={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
