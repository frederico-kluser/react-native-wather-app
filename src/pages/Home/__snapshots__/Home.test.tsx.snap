// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 16,
          "marginBottom": 32,
          "paddingLeft": 16,
          "paddingRight": 48,
          "shadowColor": "rgba(0, 0, 0, 0.24)",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 8,
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Digite o nome da cidade"
      quandity={0}
      style={
        Array [
          Object {
            "height": 48,
            "width": "100%",
          },
        ]
      }
      value=""
    />
  </View>
</View>
`;
